Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -133,7 +133,7 @@
 
   // Do not wait for server response under lock. Do not take read/write action under lock.
   private final Object myLock = new Object();
-  @Nullable private AnalysisServer myServer;
+  @Nullable private RemoteAnalysisServerImpl myServer;
```

Comment:
I'm curious why references to `AnalysisServer` were changed to `RemoteAnalysisServerImpl`? Shouldn't we use the more generic interface? Will this have implications for clients who use this as API (the Flutter plugin)?

Reply:
The reason changed it was do to the removal of `generateUniqueId()` and 2 `sendRequestToServer()` methods.  Instead of adding these to the generated AnalysisServer I as I copied them over, I changed this variable to be more specific it its type.  I don't have a strong preference if folks do think this should remain generic.

It is private and not referenced by the Flutter plugin so it won't effect plugins downstream.

Reply:
gotcha, thanks for the explaination